pyproject.toml

```diff
@@ -109,6 +109,7 @@ source = "vcs"
 [tool.pytest.ini_options]
 addopts = "-ra --strict-config --strict-markers --doctest-modules --showlocals -s -vv --durations=0 --template=maint_tools/templates/index.html -n auto"
 doctest_optionflags = "NORMALIZE_WHITESPACE ELLIPSIS"
+filterwarnings = ["error"]
```

Comment:
I started to make sure that Nilearn follows the scientific python recommendations last year (sorry too lazy to find the PR).

Turning exact warnings into errors triggers >500 to fail so I think we decided to ignore this recommendation.

Vaguely related (as in it relates to warnings):
- https://github.com/nilearn/nilearn/issues/3541
- https://github.com/nilearn/nilearn/issues/4668
- we ingore the flake8 bugbear rule related to warning stack level

I would love to see this implemented but I don't think we have quite the bandwidth at the moment: also we don't have a plan of actions on how to tackle these issues. Probably some "divide & conquer" approach working module by module and possibly having more nilearn specific warning classes.

But for the moment I would remove that pytest option: unless you are volunteering to fix all those tests? 😜

Reply:
As an example, it is expected to have deprecation warnings in tests that test deprecated functions.

These warnings should be disabled somehow. I probably need to read [How to capture warnings](https://docs.pytest.org/en/stable/how-to/capture-warnings.html) - perhaps in a different PR.

Reply:
OK then I will remove this line and merge the rest